pkg/webhook/servicecatalog/clusterservicebroker/validation/handler.go

```diff
@@ -0,0 +1,133 @@
+	Validate(context.Context, admission.Request, *sc.ClusterServiceBroker, *webhookutil.TracedLogger) *webhookutil.WebhookError
+}
+
+// AdmissionHandler handles ClusterServiceBroker validation
+type AdmissionHandler struct {
+	decoder *admission.Decoder
+
+	CreateValidators []Validator
+	UpdateValidators []Validator
+}
+
+var _ admission.Handler = &AdmissionHandler{}
+var _ admission.DecoderInjector = &AdmissionHandler{}
+var _ inject.Client = &AdmissionHandler{}
+
+// NewAdmissionHandler creates new AdmissionHandler and initializes validators list
+func NewAdmissionHandler() *AdmissionHandler {
+	return &AdmissionHandler{
+		CreateValidators: []Validator{&AccessToBroker{}},
+		UpdateValidators: []Validator{&AccessToBroker{}},
+	}
+}
+
+// Handle handles admission requests.
+func (h *AdmissionHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	traced := webhookutil.NewTracedLogger(req.UID)
+	traced.Infof("Start handling validation operation: %s for %s", req.Operation, req.Kind.Kind)
+
+	csb := &sc.ClusterServiceBroker{}
+	if err := webhookutil.MatchKinds(csb, req.Kind); err != nil {
+		traced.Errorf("Error matching kinds: %v", err)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if err := h.decoder.Decode(req, csb); err != nil {
+		traced.Errorf("Could not decode request object: %v", err)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	traced.Infof("start validation process for %s: %s/%s", csb.Kind, csb.Namespace, csb.Name)
+
+	var err *webhookutil.WebhookError
+
+	switch req.Operation {
+	case admissionTypes.Create:
+		for _, v := range h.CreateValidators {
+			err = v.Validate(ctx, req, csb, traced)
+			if err != nil {
+				break
+			}
+		}
+	case admissionTypes.Update:
+		for _, v := range h.UpdateValidators {
+			err = v.Validate(ctx, req, csb, traced)
+			if err != nil {
+				break
+			}
+		}
+	default:
+		traced.Infof("ClusterServiceBroker validation wehbook does not support action %q", req.Operation)
+		return admission.Allowed("action not taken")
+	}
+
+	if err != nil {
+		switch err.Code() {
+		case http.StatusForbidden:
+			return admission.Denied(err.Error())
+		default:
+			return admission.Errored(err.Code(), errors.New(err.Error()))
+		}
+	}
+
+	traced.Infof("Completed successfully validation operation: %s for %s: %q", req.Operation, req.Kind.Kind, req.Name)
+	return admission.Allowed("ClusterServiceBroker AdmissionHandler successful")
+}
+
+// InjectDecoder injects the decoder into the handlers
+func (h *AdmissionHandler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+
+	for _, v := range h.CreateValidators {
+		admission.InjectDecoderInto(d, v)
+	}
+	for _, v := range h.UpdateValidators {
+		admission.InjectDecoderInto(d, v)
+	}
+
+	return nil
+}
+
+// InjectClient injects the client into the handlers
+func (h *AdmissionHandler) InjectClient(c client.Client) error {
+	for _, v := range h.CreateValidators {
+		inject.ClientInto(c, v)
```

Comment:
you should check the error and return it. (return fast it's ok here) 

Same suggestion for other usecases